File: ai-form-assistant-node/src/common/repositories/baseRepository.js
class BaseRepository {
    constructor(model) {
        this.model = model;
    }

    async getAll(options = {}) {
        return this.model.findAll(options);
    }

    async getById(id, options = {}) {
        return this.model.findByPk(id, options);
    }

    async create(data, options = {}) {
        return this.model.create(data, options);
    }

    async update(id, data, options = {}) {
        const record = await this.getById(id, options);
        if (record) {
            return record.update(data, options);
        }
        return null;
    }

    async delete(id, options = {}) {
        const record = await this.getById(id, options);
        if (record) {
            await record.destroy(options);
            return true;
        }
        return false;
    }
}

module.exports = BaseRepository;

File: ai-form-assistant-fastapi/app/common/repositories/base_repository.py
from typing import Generic, Type, TypeVar, List, Optional
from sqlalchemy.orm import Session
from app.database.base import Base

ModelType = TypeVar("ModelType", bound=Base)


class BaseRepository(Generic[ModelType]):

    def __init__(self, model: Type[ModelType]):
        self.model = model

    def get_all(self, db: Session) -> List[ModelType]:
        return db.query(self.model).all()

    def get_by_id(self, db: Session, id_val) -> Optional[ModelType]:
        return db.query(self.model).filter(
            self.model.id == id_val
        ).first()

    def create(self, db: Session, obj: ModelType) -> ModelType:
        db.add(obj)
        db.flush()
        db.refresh(obj)
        return obj

    def delete(self, db: Session, obj: ModelType) -> None:
        db.delete(obj)
        db.flush()
